src/Workspaces/Core/Portable/NamingStyles/NamingStyleRules.cs

```diff
@@ -46,5 +55,33 @@ private bool IsSymbolNameAnalyzable(ISymbol symbol)
 
             return true;
         }
+
+        private bool IsInterfaceImplementation(ISymbol symbol)
+        {
+            var containingType = symbol.ContainingType;
+            if (containingType == null)
+            {
+                return false;
+            }
+
+            var implementedTypes = containingType.AllInterfaces.Concat(containingType.GetBaseTypes().ToImmutableArray());
+
+            for (var i = 0; i < implementedTypes.Length; i++)
+            {
+                var implementedType = implementedTypes[i];
+                var implementedTypeMembers = implementedType.GetMembers();
+                for (var j = 0; j < implementedTypeMembers.Length; j++)
+                {
+                    var implementedTypeMember = implementedTypeMembers[j];
+                    if (containingType.FindImplementationForInterfaceMember(implementedTypeMember) == symbol ||
+                        containingType.FindImplementationForAbstractMember(implementedTypeMember) == symbol)
```

Comment:
PR For Personal Review Only! :wink: There's a bunch of stuff I want to change. Sorry for the confusion, and thanks for the comments.